### PR TITLE
Refactoring with refactor tractor

### PR DESCRIPTION
**Pull request by tgraham777**
@rrgayhart
@selfup
#### What's this PR do?
- Restructures the shortenText() function to streamline the code and remove a variable that was being used outside of scope
- Determines the proper use for ".on" vs ".one" in each applicable case
- Improves console error messages
- Removes unnecessary whitespace
- Adds missing semicolons
- Makes styling consistent
#### Where should the reviewer start?
- Start and end in the custom.js file
#### How should this be manually tested?
- Use the browser to observe that ideas are created, edited (in-line) and deleted properly
- Observe how clicking Thumbs Up or Thumbs Down instantly changes the quality of the idea, unless the idea is already at the highest (Thumbs Up) or lowest (Thumbs Down) quality rating, in which case a console error message is thrown
#### What gif best describes this PR or how it makes you feel?

<iframe src="//giphy.com/embed/kfX3j9q5JoGkg" width="480" height="352" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="http://giphy.com/gifs/monkey-funny-kfX3j9q5JoGkg">via GIPHY</a></p>

### CODE REVIEW

**Comment by rrgayhart** [comment]
Excellent .gif! Btw, you can add images inline by using this markdown code:

``` markdown
  ![fab-u-lous monkey](http://i.giphy.com/kfX3j9q5JoGkg.gif)
```

File: app/assets/javascripts/custom.js [hunk 1]
```diff
@@ -27,85 +27,87 @@ var ideaBox = {
...
     $.ajax({
       type: 'PATCH',
       url: '/api/v1/ideas/' + $idea.attr('id'),
       data: {idea: {body: $body}},
-      success: function(){
+      success: function() {
         ideaBox.getIdeas();
       },
-      error: console.log("failed")
+      error: console.log("Body update failed")
```
Comment (rrgayhart) [inline app/assets/javascripts/custom.js:100]: Nice!

File: app/assets/javascripts/custom.js [hunk 2]
```diff
@@ -153,22 +156,23 @@ var ideaBox = {
     });
   },
 
-  shortenText: function(string){
-    if(string.length >= 100) {
+  shortenText: function(string) {
+    if(string.length > 100) {
       var cut = string.lastIndexOf(' ', 100);
-      var shortenedString =  string.substring(0, cut) + '...';
+      return string.substring(0, cut) + '...';
```
Comment (rrgayhart) [inline app/assets/javascripts/custom.js:162]: Good refactor here!

**Comment by rrgayhart** [comment]
:ship: it!

File: app/assets/javascripts/custom.js (hunk 1 above)
Comment (selfup) [inline app/assets/javascripts/custom.js:100]: Much clearer message. Dig it!

File: app/assets/javascripts/custom.js [hunk 3]
```diff
@@ -113,39 +115,40 @@ var ideaBox = {
...
-  thumbsDown: function(){
-    $('.idea-list').one('click', '#thumbs-down-button', function(){
+  thumbsDown: function() {
+    $('.idea-list').one('click', '#thumbs-down-button', function() {
       var $idea = $(this).closest('.idea');
       var $quality = $idea.find('#idea-quality').text();
       if($quality === "swill") {
-          console.log("Idea already rated as swill!")
+          console.log("Idea already rated as swill!");
+          ideaBox.getIdeas();
```
Comment (selfup) [inline app/assets/javascripts/custom.js:130]: Good call on pulling the ideas

File: app/assets/javascripts/custom.js [hunk 4]
```diff
@@ -113,39 +115,40 @@ var ideaBox = {
...
+      success: function() {
         ideaBox.getIdeas();
       },
     });
   },
 
-  truncateBody: function(){
-    $('.idea').each(function(_, idea){
+  truncateBody: function() {
+    $('.idea').each(function(_, idea) {
```
Comment (selfup) [inline app/assets/javascripts/custom.js:151]: Just some basic syntax! Feels good though

**Comment by selfup** [comment]
Effective refactor considering how much your code base has been refactored since break. Always feels good to try out new ideas and refactor syntax.

**Comment by selfup** [comment]
@rrgayhart Also you can just leave the brackets empty for gifs! ``

Seems to work just fine!

**Pull request merged**